### DISCUSSION TITLE
General Questions

### DISCUSSION THREAD

**Discussion by zlokomatic** [apache/cloudberry#456] [Q&A]
Hi,

as the greenplumdb source code vanished and i can't find any official information regarding the future of the database im looking for ways to replace the gpdb.

Maybe you can help me answer some general questions about replacing the gpdb 7 with cbdb.

- Can we use cloudberry as a drop in replacement ( so just copy the current gpdb database files ) or do we need to export / reimport the data
- Does pxf work for accessing hive? In our ETL we are accessing hive via pxf and inserting the data into gpdb
- Does PL/Java work with cbdb

Regards,

David

**Comment by RyanWei** [comment]
Hey David,

Please check the following:

> Can we use cloudberry as a drop in replacement ( so just copy the current gpdb database files ) or do we need to export / reimport the data 

No, you have to reimport data as cloudberry have different kernels as GPDB 7.

> Does pxf work for accessing hive? In our ETL we are accessing hive via pxf and inserting the data into gpdb. 

Yes, should be same as GPDB. But not open sourced yet, on the way.

> Does PL/Java work with cbdb.

Yes, should be same as GPDB. But not open sourced yet, on the way. 

You can try the current version and more features should be availale soon.

Regards,
Ryan

**Reply by zlokomatic** [reply]
> Yes, should be same as GPDB. But not open sourced yet, on the way.
So is it available in the cbdb package or how can i install it?

Do you plan to keep this project alive in the long run? Are there any sponsors to support you?

Regards,

David

**Reply by tuhaihe** [reply]
Yes, we want to keep Cloudberry running for a long time. This project is initiated and sponsored by HashData, a team built by a few original Greenplum developers, PostgreSQL contributors, and ASF Committers. We are in discussions with the ASF/Linux Foundation to explore hosting Cloudberry with them.

**Reply by zlokomatic** [reply]
Sorry my quote was bad.

> Yes, should be same as GPDB. But not open sourced yet, on the way.

Does that mean it will take some time until pxf is usable? I tried to use gpdb pxf nad it didnt work.
If so, how long will it take until pxf and PL/Java will be release for cbdb?

**Reply by tuhaihe** [reply]
> how long will it take until pxf and PL/Java will be release for cbdb?

I’m not sure, but we can high the work’s priority to public them ASAP.

**Reply by zlokomatic** [reply]
> I’m not sure, but we can high the work’s priority to public them ASAP.

That would be great

**Reply by edespino** [reply]
@zlokomatic Can you share what issues you had with PXF. I have been testing Cloudberry DB and PXF and have been successful with a few profiles (file:parquet, minio:cvs, gs:cvs). Please join the Slack instance and share what issues you are encountering and someone can try and assist. https://cloudberrydb.org/community/slack

**Comment by tuhaihe** [comment]
Hey @zlokomatic, I have some exciting news to share with you. PXF and PL/Java for Cloudberry Database are now available on GitHub:

- PXF for Cloudberry Database: https://github.com/cloudberrydb/pxf
- PL/Java for Cloudberry Database: https://github.com/cloudberrydb/pljava

We'd love to hear your thoughts, so feel free to share your feedback with us on GitHub if any problems! :blush:
